internal/commands/app/create_test.go

```diff
@@ -297,12 +300,121 @@ func TestAppCreateHandler(t *testing.T) {
 			fmt.Sprintf("  Client App ID    "+fmtStr, "test-app-abcde"),
 			"  Realm Directory  " + localApp.RootDir,
 			fmt.Sprintf("  Realm UI         "+fmtStr, "http://localhost:8080/groups/123/apps/456/dashboard"),
-			fmt.Sprintf("  Data Source      "+fmtStr, "test-cluster"),
+			fmt.Sprintf("  Data Source      "+fmtStr, "mongodb-atlas"),
 			"01:23:45 UTC DEBUG Check out your app: cd ./test-app && realm-cli app describe",
 			"",
 		}, "\n"), out.String())
 	})
 
+	t.Run("should create a minimal project dry run", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
+		profile.SetRealmBaseURL("http://localhost:8080")
+
+		out, ui := mock.NewUI()
+
+		client := mock.AtlasClient{}
+
+		cmd := &CommandCreate{
+			inputs: createInputs{
+				newAppInputs: newAppInputs{
+					Name:            "test-app",
+					Project:         "123",
+					Location:        realm.LocationVirginia,
+					DeploymentModel: realm.DeploymentModelGlobal,
+				},
+				DryRun: true,
+			},
+		}
+
+		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Atlas: client}))
+
+		expectedDir := filepath.Join(profile.WorkingDirectory, "test-app")
+		assert.Equal(t, strings.Join([]string{
+			"01:23:45 UTC INFO  A minimal Realm app would be created at " + expectedDir,
+			"01:23:45 UTC DEBUG To create this app run: " + cmd.commandString(true),
+			"",
+		}, "\n"), out.String())
+	})
+
+	t.Run("should create a dry run for the specified from app", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
+		profile.SetRealmBaseURL("http://localhost:8080")
+
+		out, ui := mock.NewUI()
+
+		testApp := realm.App{
+			ID:          "789",
+			GroupID:     "123",
+			ClientAppID: "from-app-abcde",
+			Name:        "from-app",
+		}
+
+		client := mock.RealmClient{}
+		client.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{testApp}, nil
+		}
+
+		cmd := &CommandCreate{
+			inputs: createInputs{
+				newAppInputs: newAppInputs{
+					Name:            "test-app",
+					Project:         "123",
+					Location:        realm.LocationVirginia,
+					DeploymentModel: realm.DeploymentModelGlobal,
+					From:            "from-app",
+				},
+				DryRun: true,
+			},
+		}
+
+		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: client}))
+
+		expectedDir := filepath.Join(profile.WorkingDirectory, "test-app")
+		assert.Equal(t, strings.Join([]string{
+			"01:23:45 UTC INFO  A Realm app based on the Realm app from-app would be created at " + expectedDir,
+			"01:23:45 UTC DEBUG To create this app run: " + cmd.commandString(true),
+			"",
+		}, "\n"), out.String())
+	})
+
+	t.Run("should create a minimal project dry run with data source set", func(t *testing.T) {
```

Comment:
[suggestion] seems like we could table these dry-run tests to help read them all a bit easier

---

it's also probably worth adding a quick couple of unit tests (for empty & full inputs) on the `commandString` func

Reply:
added the two suggested tests